server/src/internalClusterTest/java/org/elasticsearch/indices/cluster/ResolveClusterIT.java

```diff
@@ -0,0 +1,687 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices.cluster;
+
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.action.ActionFuture;
+import org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequest;
+import org.elasticsearch.action.admin.indices.resolve.ResolveClusterAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.test.AbstractMultiClustersTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.transport.NoSuchRemoteClusterException;
+import org.elasticsearch.transport.RemoteClusterAware;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Tests the ResolveClusterAction around all features except data streams
+ * (since those are not reachable from this package/module).
+ * ResolveClusterDataStreamIT is a sibling IT test that does additional testing
+ * related to data streams.
+ */
+public class ResolveClusterIT extends AbstractMultiClustersTestCase {
```

Comment:
Thanks for pointing it out. I didn't realize that was possible. Cross-cluster chaining actually does work or can be made to work for this endpoint.

I set up (local) -> remote1 -> remote2 and did this query:
```
GET /_resolve/cluster/*,remote1:remote2:*
```

and I logged everything coming back from the call to remote1 which was:

```
REMOTE RESPONSE from remote1: {remote2=ResolveClusterInfo{connected=true, skipUnavailable=true, matchingIndices=true, build=[tar][9e6a5bae2c088caa5188eba4249b1608a9a96616][2023-12-12T19:53:59.715578Z][8.13.0-SNAPSHOT], error=null}}
```

So this could actually be supported if we wanted to for this new endpoint.

@javanna - do we want to support cross-cluster chaining in this API?

Side note: I just discovered by experimentation that cross-cluster chaining works for CCS minimize_roundtrips=true, but not for MRT=false.

Reply:
I don't think we should worry about something that we never supported, at least not on purpose.